retis/src/core/probe/kernel/probe_stack.rs

```diff
@@ -44,11 +41,36 @@ impl ProbeStack {
 
         Self {
             probes: HashSet::from_iter(attached_probes),
-            keep_stack,
+            display_stack_cache: HashSet::new(),
             known_kernel_types,
         }
     }
 
+    fn keep_stack(&mut self, mgr: &mut ProbeRuntimeManager, evt: &KernelEvent) -> bool {
+        let r#type = match evt.probe_type.as_str() {
+            "raw_tracepoint" => "tp",
+            s => s,
+        };
+
+        let sym = format!("{}:{}", r#type, evt.symbol);
+
+        if self.display_stack_cache.contains(&sym) {
+            return true;
+        }
+
+        let keep_stack = if let Some(opts) = mgr.get_probe_opts(&sym) {
+            opts.contains(&crate::core::probe::ProbeOption::ReportStack)
+        } else {
+            false
+        };
+
+        if keep_stack {
+            self.display_stack_cache.insert(sym);
+        }
```

Comment:
This means each time a probe is not reporting a stack trace, we'll do the lookup for each event? We should cache both true and false values to have a 100% hit.

Reply:
I honestly see no value in caching here. I added it right after pushing. (How many options does a probe have to make the caching worth it?)

Reply:
we rather have a double lookup in the worst case

Reply:
also caching makes things less dynamic. What if the runtime starts removing probes? What if something cached in the false becomes true because it gets added with that option?

Reply:
Right, if the probe options are changed at runtime or removed, we would have incorrect information. This is not supported ATM by the manager and would require heavy changes though.

I don't have strong opinion wrt to caching vs not caching. I guess we're looking at a single local lookup vs a lookup + iter + check. Idk if the difference in performances is noticeable.

Reply:
> Right, if the probe options are changed at runtime or removed, we would have incorrect information. This is not supported ATM by the manager and would require heavy changes though.
>
> I don't have strong opinion wrt to caching vs not caching. I guess we're looking at a single local lookup vs a lookup + iter + check. Idk if the difference in performances is noticeable.

it's 1 lookup + iter (constant though) vs [3 lookup (worst case); 1 lookup best case (let's say for the true case); 2 lookups (for the false case)]

The point is that the constant iteration is probably negligible (if we have tons of stacktraces, we're going to have a performance drop for other reasons)

Reply:
Makes sense, let's remove the cache.